near-contract-standards/src/fungible_token/macros.rs

```diff
@@ -107,3 +107,77 @@ macro_rules! impl_fungible_token_storage {
         }
     };
 }
+
+/// Base fungible token + pausable functionality
+#[macro_export]
+macro_rules! impl_pausable_fungible_token_core {
+    ($contract: ident, $token: ident $(, $on_tokens_burned_fn:ident)?) => {
+        use $crate::fungible_token::core::FungibleTokenCore;
+        use $crate::fungible_token::resolver::FungibleTokenResolver;
+        use $crate::fungible_token::pausable::Pausable;
+
+        #[near_bindgen]
+        impl FungibleTokenCore for $contract {
+            #[payable]
+            fn ft_transfer(
+                &mut self,
+                receiver_id: AccountId,
+                amount: U128,
+                memo: Option<String>,
+                ) {
+                assert!(!self.paused, "Contract Is Paused");
```

Comment:
From what I've seen in `near-contract-standards` lib, `require!` seems more standard for checking these things.